platform/security/filter/security-filter-csrf/src/test/groovy/org/codice/ddf/security/filter/csrf/CsrfFilterSpec.groovy

```diff
@@ -106,38 +154,30 @@ class CsrfFilterSpec extends Specification {
-        "/search/catalog/internal" | "https://example.com" | null                  | false
-        "/search/catalog/internal" | null                  | "https://example.com" | false
-        "/search/catalog/internal" | "https://example.com" | "https://example.com" | false
-        // Web sockets endpoint requiring same origin but no CSRF header -
-        "/search/catalog/ws"       | "https://example.com" | null                  | true
-        "/search/catalog/ws"       | null                  | "https://example.com" | true
-        "/search/catalog/ws"       | "https://example.com" | "https://example.com" | true
-        // Corrupted origin/referer headers
-        "/admin/jolokia"           | "com?ht.p://*example" | null                  | true
-        "/admin/jolokia"           | null                  | "com?ht.p://*example" | true
-        "/admin/jolokia"           | "com?ht.p://*example" | "com?ht.p://*example" | true
-        "/admin/jolokia"           | "!@#\$%^&*(){}:<>?"   | "!@#\$%^&*(){}:<>?"   | true
+        [requestContext, originHeader, refererHeader, hasCsrfHeader] << [
+                // Protected Contexts - no CSRF Header
+                ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
+                [null, "", EXTERNAL_SITE, DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, PROXY_HTTP_NOPORT, PROXY_HTTPS_NOPORT, DDF_BADPORT, PROXY_BADPORT],
+                [null, "", EXTERNAL_SITE, DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, PROXY_HTTP_NOPORT, PROXY_HTTPS_NOPORT, DDF_BADPORT, PROXY_BADPORT],
+                [false]
+        ].combinations() + [
+                // Protected Contexts - different or no origin/referer, with CSRF header
+                ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
+                [null, "", EXTERNAL_SITE, DDF_BADPORT, PROXY_BADPORT],
+                [null, "", EXTERNAL_SITE, DDF_BADPORT, PROXY_BADPORT],
+                [true]
+        ].combinations() + [
+                // Websockets - different or no origin/referer, with/without CSRF header
+                ["/search/catalog/ws", "/search/catalog/ws/subdirectory"],
+                [null, "", EXTERNAL_SITE, DDF_BADPORT, PROXY_BADPORT],
+                [null, "", EXTERNAL_SITE, DDF_BADPORT, PROXY_BADPORT],
+                [true, false]
+        ].combinations() + [
+                // Corrupted origin/referer headers
+                ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
+                [null, "", "com?ht.p://*example", "!@#\$%^&*(){}:<>?", "undefined", "0", "true", "\r\n", "\\r\\n"],
+                [null, "", "com?ht.p://*example", "!@#\$%^&*(){}:<>?", "undefined", "0", "true", "\r\n", "\\r\\n"],
+                [true, false]
+        ].combinations()
     }
 }
```

Comment:
Should we add a newline?????

Reply:
After much deliberation, this issue has been successfully resolved.